collection.go

```diff
@@ -88,6 +88,43 @@ func (c *LoggerCollection) getLogger(name string, level LogLevel,
 	return logger
 }
 
+// ConfigureLoggers configures loggers according to the given string
+// specification, which specifies a set of loggers and their associated
+// logging levels.  Loggers are semicolon-separated; each
+// configuration is specified as <logger>=<level>.  White space outside of
+// logger names and levels is ignored.  The default level is specified
+// with the name "DEFAULT".
+//
+// An example specification:
+//	`<root>=ERROR; foo.bar=WARNING`
+func (c *LoggerCollection) ConfigureLoggers(specification string) error {
+	confs := strings.Split(strings.TrimSpace(specification), ";")
+	for i := range confs {
+		conf := strings.SplitN(strings.TrimSpace(confs[i]), "=", 2)
+		levelstr := conf[1]
+		name := conf[0]
+		level, err := LevelFromString(levelstr)
+		if err != nil {
+			return err
+		}
+		if name == "DEFAULT" {
+			c.level = level
+			continue
+		}
+		logger, exists := c.loggers[name]
+		if !exists {
+			logger = &Logger{level: level,
```

Comment:
why create a logger? if it doesn't exist by flag parse time, we assume in other places nothing will log to it. check out how SetLevel works

Reply:
A new log can be created at any time. The convention is package init, but one can create a new logger within a function that isn't called until after flags are parsed. In this case, the config flag parsing needs to create the logger at the specified level so that when it is fetched with GetLoggerNamed in a function, it logs at the specified level.

Reply:
in that case, the code from GetLoggerNamed should be abstracted and this should just call it. i'm anxious about having multiple struct creation points

Reply:
Yes, this is much better now. Thanks.